### PR TITLE
feat: Added FilterTemplate

### PR DESCRIPTION
**Pull request by manuelelucchi**
### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

As suggested in #3283, I implemented a template for filters in table columns

### 💡 Background and solution

The problem was that the current implementation doesn't allow to customize the filters UI, after the changes you can now use `FilterTemplate` in `Column` to add your own UI

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Added `FilterTemplate` property on `Column` and `PropertyColumn` to customize the filters dropdown       |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

**Comment by manuelelucchi** [comment]
This is my first contribution so I probably did something not in the correct way. I don't know Chinese so i'm not able to add the Chinese changelog/documentation update. Also for some reason the docs didn't build without commenting the NavMenu

**Comment by manuelelucchi** [comment]
@dotnet-policy-service agree

**Comment by ElderJames** [comment]
Thank you for contribution @manuelelucchi , please checkout the review comments.

**Comment by manuelelucchi** [comment]
Sorry @ElderJames but I don't see any review comment for some reason

**Comment by ElderJames** [comment]
@manuelelucchi Sorry, I haven't submit the review, now it done.

File: components/table/Column.razor [hunk 1]
```diff
@@ -193,119 +193,126 @@ else if (IsBody && RowSpan != 0 && ColSpan != 0)
...
+									    @if (_columnFilterType == TableFilterType.FieldType)
+									    {
+										    <Button Size="small" Icon="plus" Type="primary" OnClick="AddFilter">
+										    </Button>
+									    }
+									    <Button Size="small" Type="primary" OnClick="() => FilterConfirm()">
+										    @Table?.Locale.FilterConfirm
+									    </Button>
+								    </div>
+                                }
```
[the remark this answers is not in the document]
Comment (manuelelucchi) [reply components/table/Column.razor:315]: I'm not sure about the indentation, the file is half spaces half tabs

[the remark this answers is not in the document]
Comment (ElderJames) [reply components/table/Column.razor:315]: Just comit your changes.

**Comment by manuelelucchi** [comment]
I think I have done but i'm not really able to test since even cleaning all the new sample won't appear in the docs. Also, the docs won't compile unless I remove the NavMenu for some reason

**Comment by ElderJames** [comment]
You need to rebuild the AntDesign.Docs.Build project to  generate the demo metadata into AntDesign.Docs.
The NavMenu issue is caused bythe VS 17.6, I think it was a bug of the lastest SDK.

**Comment by manuelelucchi** [comment]
> You need to rebuild the AntDesign.Docs.Build project to generate the demo metadata into AntDesign.Docs
I tried cleaning the project, deleting bin/obj, npm install npm start but it doesn't work

**Comment by ElderJames** [comment]
Ok,I will check it out.

**Comment by manuelelucchi** [comment]
Any news on this @ElderJames? I think it's an important feature for the most important component

**Comment by ElderJames** [comment]
Hello @manuelelucchi , I would merge this feature later because I'm refactoring the Table Component in #3321 , and Column would be changed.

**Review by ElderJames — APPROVED** [review]

**Pull request merged**